csrc/flash_attn/src/flash_fwd_kernel.h

```diff
@@ -854,14 +859,16 @@ inline __device__ void compute_attn_1rowblock_splitkv(const Params &params, cons
 
                 }
             }
-            tKgK.data() = tKgK.data() + (-int(kBlockN * params.k_row_stride));
-            tKgKnew.data() = tKgKnew.data() + (-int(kBlockN * params.knew_row_stride));
+            tKgK.data() = tKgK.data() + binfo.k_advance_offset_pg(bidb_cache, n_block, params.k_row_stride, kBlockN);
+            tKgKnew.data() = tKgKnew.data() + binfo.k_advance_offset_pg(bidb_cache, n_block, params.v_row_stride, kBlockN);
         }
         // Need this before we can read in K again, so that we'll see the updated K values.
         __syncthreads();
         if (n_block_max > n_block_copy_min) {
-            tKgK.data() = tKgK.data() + (n_block_max - n_block_copy_min) * kBlockN * params.k_row_stride;
-            tVgV.data() = tVgV.data() + (n_block_max - n_block_copy_min) * kBlockN * params.v_row_stride;
+            tKgK.data() = tKgK.data() - (n_block_max - n_block_copy_min) * binfo.k_advance_offset_pg(bidb_cache, n_block, params.k_row_stride, kBlockN);
```

Comment:
This code goes to the end of the block again so that we can run attention block by block  from the end of the blocks